### PR TITLE
Do not report only async gap frames for logging calls

### LINKED ISSUE #1392: All events captured by the logging integration are grouped together
### Platform

Flutter Mobile

### Obfuscation

Disabled

### Debug Info

Enabled

### Doctor

[!] Flutter (Channel stable, 3.7.1, on macOS 13.1 22C65 darwin-x64, locale en-GB)
    • Flutter version 3.7.1 on channel stable at /Users/barabas/source/flutter
    ! Warning: `dart` on your path resolves to /usr/local/Cellar/dart/2.19.3/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/barabas/source/flutter. Consider adding /Users/barabas/source/flutter/bin
      to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7048ed95a5 (3 months ago), 2023-02-01 09:07:31 -0800
    • Engine revision 800594f1f4
    • Dart version 2.19.1
    • DevTools version 2.20.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/barabas/Library/Android/sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[!] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.74.3)
    • VS Code at /Users/barabas/Downloads/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • iPhone XR (mobile) • 00008020-001E2C881A33002E • ios        • iOS 15.6.1 19G82
    • macOS (desktop)    • macos                     • darwin-x64 • macOS 13.1 22C65 darwin-x64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 3 categories.

### Version

6.22.0

### Steps to Reproduce

Enable the logging integration and write a severe log. Do not provide an error or stacktrace parameter to the logger.

### Expected Result

The event generated by the log includes a stack trace, filename and line number information of where the log was written from.

Every time the log runs, a new event is added to the same group in the web UI.

Every time a different line logs an event a new group is created for that log.

### Actual Result

The generated event has a filename and stacktrace of `<asynchronous suspension>`. All of the severe logs in the project are grouped under this stack trace in the web UI using the default grouping.

image

image

I expected every instance of a severe log to be grouped with other events generated by the same line. I tried to set up a custom group, but the best I seem to be able to do is to completely prevent grouping using the following line:

`stack.abs_path:"<asynchronous suspension>" -> nostack, {{ message }}`

### Are you willing to submit a PR?

Yes

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
Also upgraded the tooling to compile with Java 17
https://github.com/getsentry/sentry-dart/pull/1399 was merged into this PR.

## :bulb: Motivation and Context
Fixes https://github.com/getsentry/sentry-dart/issues/1392

## :pencil: Checklist

- [X] I reviewed submitted code
- [X] I added tests to verify changes
- [X] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [ ] I updated the docs if needed
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

File: flutter/example/lib/main.dart [hunk 1]
```diff
@@ -394,6 +394,13 @@ class MainScaffold extends StatelessWidget {
               },
               child: const Text('Show UserFeedback Dialog without event'),
             ),
+            ElevatedButton(
+              onPressed: () {
+                final log = Logger('Logging');
+                log.info('My Logging test 2');
```
Comment (marandaneto) [inline flutter/example/lib/main.dart]:
```suggestion
                log.info('My Logging test');
```

File: dart/test/stack_trace_test.dart [hunk 2]
```diff
@@ -249,6 +249,21 @@ isolate_instructions: 10fa27070, vm_instructions: 10fa21e20
...
+          .getStackFrames(StackTrace.fromString('''
+#0      SentryClient._prepareEvent (package:sentry/src/sentry_client.dart:206:33)
+#1      SentryClient.captureEvent (package:sentry/src/sentry_client.dart:74:34)
+#2      Hub.captureEvent (package:sentry/src/hub.dart:97:38)
+<asynchronous suspension>
+#3      LoggingIntegration._onLog (package:sentry_logging/src/logging_integration.dart:58:7)
+<asynchronous suspension>
+            '''))
+          .map((frame) => frame.toJson());
+      expect(frames.isEmpty, true);
```
Comment (krystofwoldrich) [inline dart/test/stack_trace_test.dart:265]: It looks good, I just lack context on why we want to remove all frames when the async gap is left?

Comment (krystofwoldrich) [reply dart/test/stack_trace_test.dart:265]: NVM I read the issue now i get it.
https://github.com/getsentry/sentry-dart/issues/1392

**Review by krystofwoldrich — APPROVED** [review]
Looks good. Lot of things improved and fixed. Nice. 🚀

**Pull request merged**